Week1/github.js

```diff
@@ -0,0 +1,93 @@
+        containerForContributors.innerHTML = "";
+        const contributorsURL = repository['contributors_url'];
+        createAndAppend('p', containerForContributors, { text: 'Contributions' });
+        fetchJSON(contributorsURL)
+            .catch(err => {
+                createAndAppend('div', containerForContributors, { text: err.message, class: 'alert-error' });
+              })
+
+              .then(contributors => {
+                const contributorsTable = createAndAppend('table', containerForContributors, { id: 'contributorsTable' });
+
+                const contributorsTbody = createAndAppend('tbody', contributorsTable, { id: 'contributorsTbody' });
+               
+                contributors.forEach(contributor => {
+
+                    const contLink = contributor.html_url;
+                    const contImg = contributor.avatar_url;
+                    const contName = contributor.login;
+                    const contNum = contributor.contributions;
+                    const contributorsLink = createAndAppend('a', contributorsTbody, { href: contLink, target: '_blank', class: 'contributorsLink' });
+
+                const contributorsTr = createAndAppend('tr', contributorsLink, { id: `${contributor['id']}`, class: 'contributorsTr' });
+
+                const tdImg = createAndAppend('td', contributorsTr, { id: `${contName}Td`, class: 'inform-contrib' });
+
+                createAndAppend('img', tdImg, { src: contImg, alt: `picture of ${contName}`, class: 'contributor-images' });
+
+                createAndAppend('td', contributorsTr, { text: contName, class: 'inform-contrib contributor-name' });
+
+                createAndAppend('td', contributorsTr, { text: contNum, class: 'inform-contrib contributor-number' });
+            });
+        })
+    }
+    
+
+    function main(url) {
+       const root = document.getElementById('root');
+       const divForSelect = createAndAppend('div', root, {id: "container-select"});
+       createAndAppend('h5', divForSelect, {text: "HYF Repositories", class: "heading"});
+       const select = createAndAppend('select', divForSelect, {id: "menu"});
+       const container = createAndAppend('div', root, {id: "first-container"});
+       const containerForContributors = createAndAppend('div', root, {id: "second-container"})
+
+    fetchJSON(url)
+        .catch(err => {
+            container.innerHTML = error.message;
+            return;
+        })
+    
+        .then(repositories => {
+        
+            repositories.sort((a, b) => a.name.localeCompare(b.name));
+
+            repositories.forEach((repository, index) => {
+            createAndAppend('option', select, { text: repository.name, value: index });
+        });
+        
+        select.addEventListener('change', (event) => {
+            renderRepository(container, repositories[event.target.value])
+            renderContributors(containerForContributors, repositories[event.target.value])
+        });
```

Comment:
You will notice that there is a slight delay between showing the repo info and the contributors info. Time-wise it looks like this.

[show repo][clear contributors container][fetch contributors].......[show contributors]

You can improve on the user experience by postponing the (re-)rendering until the contributor data has been returned:

[fetch contributors].......[clear contributors container][render contributors][show repo]

Now, while the contributors are being fetched the screen remains unchanged.

To achieve this in your code, swap the two lines as follows:

```js
renderContributors(containerForContributors, repositories[event.target.value]);
renderRepository(container, repositories[event.target.value]);
```

Then, in your `renderContributors()` function, postpone clearing out the `containerForContributors` until after you have received the response from the network:

```js
function renderContributors(containerForContributors, repository) {
  const contributorsURL = repository['contributors_url'];
  createAndAppend('p', containerForContributors, { text: 'Contributions' });
  fetchJSON(contributorsURL)
    .then(contributors => {
      containerForContributors.innerHTML = "";
     //...
```